internal/dbadapter/db_adapter.go

```diff
@@ -377,66 +352,53 @@ func (a *StandardAdapter) JobGetAvailableTx(ctx context.Context, tx pgx.Tx, queu
 }
 
-func (a *StandardAdapter) JobSetCompletedTx(ctx context.Context, tx pgx.Tx, id int64, completedAt time.Time) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetCompleted(ctx, tx, dbsqlc.JobSetCompletedParams{
-		ID:          id,
-		FinalizedAt: completedAt.UTC(),
-	})
+func JobSetStateCompleted(id int64, finalizedAt time.Time) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, finalizedAt: &finalizedAt, state: dbsqlc.JobStateCompleted}
 }
 
-func (a *StandardAdapter) JobSetDiscardedIfRunning(ctx context.Context, id int64, finalizedAt time.Time, err []byte) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetDiscardedIfRunning(ctx, a.executor, dbsqlc.JobSetDiscardedIfRunningParams{
-		ID:          id,
-		Error:       err,
-		FinalizedAt: finalizedAt,
-	})
+func JobSetStateDiscarded(id int64, finalizedAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, finalizedAt: &finalizedAt, state: dbsqlc.JobStateDiscarded}
 }
 
-func (a *StandardAdapter) JobSetErroredIfRunning(ctx context.Context, id int64, scheduledAt time.Time, err []byte) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
+func JobSetStateErrored(id int64, scheduledAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, scheduledAt: &scheduledAt, state: dbsqlc.JobStateRetryable}
+}
 
-	return a.queries.JobSetErroredIfRunning(ctx, a.executor, dbsqlc.JobSetErroredIfRunningParams{
-		ID:          id,
-		Error:       err,
-		ScheduledAt: scheduledAt,
-	})
+func JobSetStateSnoozed(id int64, scheduledAt time.Time, maxAttempts int) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, maxAttempts: &maxAttempts, scheduledAt: &scheduledAt, state: dbsqlc.JobStateScheduled}
 }
 
-func (a *StandardAdapter) JobSetSnoozedIfRunning(ctx context.Context, id int64, scheduledAt time.Time) (*dbsqlc.RiverJob, error) {
+func (a *StandardAdapter) JobSetState(ctx context.Context, params *JobSetStateParams) (*dbsqlc.RiverJob, error) {
 	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
 	defer cancel()
 
-	return a.queries.JobSetSnoozedIfRunning(ctx, a.executor, dbsqlc.JobSetSnoozedIfRunningParams{
-		ID:          id,
-		ScheduledAt: scheduledAt,
+	return a.queries.JobSetStateIfRunning(ctx, a.executor, dbsqlc.JobSetStateIfRunningParams{
+		ID:                  params.ID,
+		ErrorDoUpdate:       params.errData != nil,
+		Error:               params.errData,
+		FinalizedAtDoUpdate: params.finalizedAt != nil,
+		FinalizedAt:         params.finalizedAt,
+		MaxAttemptsUpdate:   params.maxAttempts != nil,
+		MaxAttempts:         int16(min(ptrutil.ValOrDefault(params.maxAttempts, 0), math.MaxInt16)), // default never used
```

Comment:
This line is confusing me a bit. What is the `min` achieving?

Reply:
I actually put this in because I thought you'd want it :) Basically, it's protecting against a possible overflow as `int` becomes `int16`.

That said, this will never actually be a problem in practice, so I don't care if it's there or not. Took it out.